src/consensus/aft/raft.h

```diff
@@ -141,8 +131,10 @@ namespace aft
     size_t sig_tx_interval;
 
     // Configurations
-    std::list<Configuration> configurations;
-    std::unordered_map<ccf::NodeId, NodeState> nodes;
+    aft::ConfigurationTracker configuration_tracker;
+    std::unordered_map<ccf::NodeId, aft::NodeState> nodes;
+    bool catching_up = false;
+    bool retiring = false;
```

Comment:
I'll put up a separate PR for this, but I think a flag isn't enough and we need to keep track of more than that to ensure a single signature retirement.

Reply:
Yes, retirement changes aren't implemented here, because I thought we wouldn't need it. We'll need a lot more to do that properly.